tests/ui/feature-gates/feature-gate-explicit-extern-abis.rs

```diff
@@ -0,0 +1,30 @@
+// This test differs from other feature gate tests.
+// Instead of verifying that an error occurs when the feature gate is missing,
+// it ensures that only a warning is emitted, not an error.
```

Comment:
```suggestion
// The purpose of this feature gate is to make something into a hard error in a
// future edition.  Consequently, this test differs from most other feature gate
// tests.  Instead of verifying that an error occurs when the feature gate is
// missing, it ensures that the hard error is only produced with the feature
// gate is present in the `future` edition -- and otherwise that only a warning
// is emitted.
```